PhysObjectExtractor/src/ElectronAnalyzer.cc

```diff
@@ -114,30 +121,94 @@ ElectronAnalyzer::analyze(const edm::Event& iEvent, const edm::EventSetup& iSetu
 void
 ElectronAnalyzer::analyzeElectrons(const edm::Event& iEvent, const edm::Handle<reco::GsfElectronCollection> &electrons)
 {
-	  numelectron = 0;
-	  electron_e.clear();
-	  electron_pt.clear();
-	  electron_px.clear();
-	  electron_py.clear();
-	  electron_pz.clear();
-	  electron_eta.clear();
-	  electron_phi.clear();
-	  electron_ch.clear();
-
+  using namespace edm;
+  using namespace std;
+  edm::Handle<reco::ConversionCollection> hConversions;
+  iEvent.getByLabel("allConversions", hConversions);
+  edm::Handle<reco::BeamSpot> bsHandle;
+  iEvent.getByLabel("offlineBeamSpot", bsHandle);
+  const reco::BeamSpot &beamspot = *bsHandle.product();
+  Handle<reco::VertexCollection> vertices;
+  iEvent.getByLabel(InputTag("offlinePrimaryVertices"), vertices);
+  math::XYZPoint pv(vertices->begin()->position());
+
+  numelectron = 0;
+  electron_e.clear();
+  electron_pt.clear();
+  electron_px.clear();
+  electron_py.clear();
+  electron_pz.clear();
+  electron_eta.clear();
+  electron_phi.clear();
+  electron_ch.clear();
+  electron_isLoose.clear();
+  electron_isMedium.clear();
+  electron_isTight.clear();
+  
   if(electrons.isValid()){
      // get the number of electrons in the event
-     numelectron=electrons->size();
-        for (reco::GsfElectronCollection::const_iterator itElec=electrons->begin(); itElec!=electrons->end(); ++itElec){
-
-    	        electron_e.push_back(itElec->energy());
-    	        electron_pt.push_back(itElec->pt());
-    	        electron_px.push_back(itElec->px());
-    	        electron_py.push_back(itElec->py());
-    	        electron_pz.push_back(itElec->pz());
-    	        electron_eta.push_back(itElec->eta());
-    	        electron_phi.push_back(itElec->phi());
-    	        electron_ch.push_back(itElec->charge());
+    numelectron=electrons->size();
+    float pfIso = -999;
+    for (reco::GsfElectronCollection::const_iterator itElec=electrons->begin(); itElec!=electrons->end(); ++itElec){
+      int missing_hits = itElec->gsfTrack()->trackerExpectedHitsInner().numberOfHits()-itElec->gsfTrack()->hitPattern().numberOfHits();
+      bool passelectronveto = !ConversionTools::hasMatchedConversion(*itElec, hConversions, beamspot.position());
+      if (itElec->passingPflowPreselection()) {
+        auto iso03 = itElec->pfIsolationVariables();
+        float pfIso = (iso03.chargedHadronIso + iso03.neutralHadronIso + iso03.photonIso)/itElec->pt();
```

Comment:
Here's a definition as a float of the pfIso variable, but in line 151 there another definition (with an original declaration).  This is giving warning at compilation that are being treated as errors.  @npervan could you please look at that?

Reply:
```suggestion
       pfIso = (iso03.chargedHadronIso + iso03.neutralHadronIso + iso03.photonIso)/itElec->pt();
```

Reply:
Hi Edgar--sorry for the delay. I'll fix both of these this week

Reply:
When you say new variables do you just mean the isLoose, isMedium, isTight variables (which are missing and need to be written to the file). Or would you also like me to write the input variable (pfIso, etc) to the root files?

Reply:
Hi @npervan.  No problem, and thanks! 
I suppose I meant the isLoose, isMedium, etc.  I.e., the ones for which you book a container, along with the other kinematics variables, but do not write them yet (I had the impression that you meant to do that).  For the other ones, if they are just intermediate steps, there is no need.